src/config.js

```diff
@@ -63,6 +67,23 @@ function debug () {
   if (HEROKU_DEBUG) return parseInt(HEROKU_DEBUG)
   return 0
 }
+function skipAnalytics () {
+  if (process.env['TESTING'] && process.env['TESTING'].match(/[\w]+/)) {
+    return true
+  } else if (this._skipAnalytics) {
+    return true
+  } else if (this.netrcLogin() === false) {
+    return true
+  }
+  return false
+}
+
+async function netrcLogin () {
```

Comment:
@KarateCowboy could you move this back to the `cli-engine` and just have the responsibility of the config to check if the `TESTING` env var is set and checking the [user config file](https://github.com/heroku/cli/blob/master/config.go#L8) for which I cannot find the implementation.